### PR TITLE
Optimize RTE page

### PR DESCRIPTION
**Pull request by sypets**
Main point it to add examples for setting a preset via page TSconfig.

Additionally, the following changes are made:

- add a header for the examples (since examples are now extended and this improves finding the examples when skimming the page)
- add clarification what "order" (for the presets) means. If not familiar with TYPO3 one might assume, the order means that the later settings overrides the previous which is not the case: once a setting is set, the ones following it are ignored.
- clarify what default (number 4. in the listing of "order") means
- add some links

----

**Note to reviewers:**

* I have not set the backport labels. I think this can be backported as is, but should double check
* Please check if this actually improves the clarity. I found some parts to be a little vague but adding extra text always has the problem that it clutters up the page and makes it less terse which might even impact the clarity negatively

### CODE REVIEW

File: Documentation/PageTsconfig/Rte.rst [hunk 1]
```diff
@@ -13,12 +13,15 @@ The `RTE` prefix key is used for configuration of the Rich Text Editor.
 Please refer to the :ref:`RTE chapter <t3coreapi:rte>` in Core API document
 for more general information on RTE configuration and data processing.
 
-The order in which configuration for the RTE is loaded is:
+The order in which configuration for the RTE is loaded is (the first one which
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:16]:
```suggestion
The order in which the configuration for the RTE is loaded is (the first one which
```

File: Documentation/PageTsconfig/Rte.rst [hunk 2]
```diff
@@ -13,12 +13,15 @@ The `RTE` prefix key is used for configuration of the Rich Text Editor.
 Please refer to the :ref:`RTE chapter <t3coreapi:rte>` in Core API document
 for more general information on RTE configuration and data processing.
 
-The order in which configuration for the RTE is loaded is:
+The order in which the configuration for the RTE is loaded is (the first one which
+is set, is used, see :ref:`example <pageTsRteOverridePreset>` below):
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:17]:
```suggestion
is set will be used, see :ref:`example <pageTsRteOverridePreset>` below):
```

File: Documentation/PageTsconfig/Rte.rst [hunk 3]
```diff
@@ -13,12 +13,15 @@ The `RTE` prefix key is used for configuration of the Rich Text Editor.
 Please refer to the :ref:`RTE chapter <t3coreapi:rte>` in Core API document
 for more general information on RTE configuration and data processing.
 
-The order in which configuration for the RTE is loaded is:
+The order in which the configuration for the RTE is loaded is (the first one which
+is set will be used, see :ref:`example <pageTsRteOverridePreset>` below):
 
 1. preset defined for a specific field via PageTS
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:19]:
```suggestion
1. preset defined for a specific field via page TSconfig
```

File: Documentation/PageTsconfig/Rte.rst [hunk 4]
```diff
@@ -13,12 +13,15 @@ The `RTE` prefix key is used for configuration of the Rich Text Editor.
...
+is set will be used, see :ref:`example <pageTsRteOverridePreset>` below):
 
-1. preset defined for a specific field via PageTS
-2. richtextConfiguration defined for a specific field via TCA
-3. general preset defined via PageTS
-4. default
+1. preset defined for a specific field via page TSconfig
+2. :ref:`richtextConfiguration <t3tca:columns-text-properties-richtextConfiguration>`
+   defined for a specific field via TCA
+3. general preset defined via PageTS (:typoscript:`RTE.default.preset`)
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:22]:
```suggestion
3. general preset defined via page TSconfig (:typoscript:`RTE.default.preset`)
```

File: Documentation/PageTsconfig/Rte.rst [hunk 5]
```diff
@@ -33,7 +36,17 @@ Configure RTE for a specific field in a table
 Configure RTE for a specific field in a table for a specific :ref:`record type <t3tca:types>`
     `RTE.config.[tableName].[fieldName].types.[type]`
 
-Consider the following Page TSconfig examples:
+Configuring RTE via page TSconfig is general and not specific to a
+specific RTE editor, however TYPO3 comes with EXT:rte_ckeditor, so this
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:40]:
```suggestion
particular RTE editor, however TYPO3 comes with EXT:rte_ckeditor, so this
```

avoiding specific after specific in the same sentence.

File: Documentation/PageTsconfig/Rte.rst [hunk 6]
```diff
@@ -33,7 +36,17 @@ Configure RTE for a specific field in a table
 Configure RTE for a specific field in a table for a specific :ref:`record type <t3tca:types>`
     `RTE.config.[tableName].[fieldName].types.[type]`
 
-Consider the following Page TSconfig examples:
+Configuring RTE via page TSconfig is general and not specific to a
+particular RTE editor, however TYPO3 comes with EXT:rte_ckeditor, so this
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:40]:
```suggestion
particular rich-text editor, however TYPO3 comes with EXT:rte_ckeditor, so this
```

RTE editor -> E means already editor.

File: Documentation/PageTsconfig/Rte.rst [hunk 7]
```diff
@@ -33,7 +36,17 @@ Configure RTE for a specific field in a table
 Configure RTE for a specific field in a table for a specific :ref:`record type <t3tca:types>`
     `RTE.config.[tableName].[fieldName].types.[type]`
 
-Consider the following Page TSconfig examples:
+Configuring RTE via page TSconfig is general and not specific to a
+particular rich-text editor, however TYPO3 comes with EXT:rte_ckeditor, so this
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:40]:
```suggestion
particular rich-text editor. However, TYPO3 comes with EXT:rte_ckeditor, so this one
```

File: Documentation/PageTsconfig/Rte.rst [hunk 8]
```diff
@@ -33,7 +36,17 @@ Configure RTE for a specific field in a table
 Configure RTE for a specific field in a table for a specific :ref:`record type <t3tca:types>`
     `RTE.config.[tableName].[fieldName].types.[type]`
 
-Consider the following Page TSconfig examples:
+Configuring RTE via page TSconfig is general and not specific to a
+particular rich-text editor. However, TYPO3 comes with EXT:rte_ckeditor, so this one
+will usually be used. This page covers only the general configuration, for
+more information about configuring rte_ckeditor, see the
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:42]:
```suggestion
more information about configuring EXT:rte_ckeditor, see the
```

File: Documentation/PageTsconfig/Rte.rst [hunk 9]
```diff
@@ -59,6 +72,30 @@ Consider the following Page TSconfig examples:
    # But disable RTE for tt_content bodytext again if the record type is "text"
    RTE.config.tt_content.bodytext.types.text.disabled = 1
 
+.. _pageTsRteOverridePreset:
+
+Example: Override preset
+------------------------
+
+Check the description of the order above to see which setting overrides which.
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:80]:
```suggestion
Refer to the description of the order above for details of which setting has priority over which.
```

File: Documentation/PageTsconfig/Rte.rst [hunk 10]
```diff
@@ -59,6 +72,30 @@ Consider the following Page TSconfig examples:
...
+.. _pageTsRteOverridePreset:
+
+Example: Override preset
+------------------------
+
+Refer to the description of the order above for details of which setting has priority over which.
+
+Summary:
+
+*   Setting the preset via page TSconfig *for a specific field* overrides all
```
Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:84]: I don't understand that sentence: "... overrides all else"

Comment (sypets) [reply Documentation/PageTsconfig/Rte.rst:84]: If set, this is the setting which will be used, regardless of any other setting

Comment (brotkrueml) [inline Documentation/PageTsconfig/Rte.rst:84]:
```suggestion
*   Setting the preset via page TSconfig *for a specific field* overrides all,
```

Comment (brotkrueml) [reply Documentation/PageTsconfig/Rte.rst]: Added a comma, to make it clearer.

**Comment by brotkrueml** [comment]
@sypets Thanks for your work :+1:

**Pull request merged**